lib/pbench/test/unit/agent/task/test_tools_kill.py

```diff
@@ -191,8 +192,31 @@ def iterdir(self):
             and pairs[1][1] == "ghijkl"
         )
 
+    @staticmethod
+    def test_gen_host_names_no_tool_groups(monkeypatch):
+        """Verify kill.gen_host_names() is a no-op when we have no tool groups
+        present.
+        """
+
+        def mock_gen_tool_groups(pbench_run: str) -> Iterable[ToolGroup]:
+            """Override kill.gen_tool_groups definition to yield no ToolGroup
+            objects.
+            """
+            for hosts in []:
```

Comment:
nit: "host"

In addition, `ToolGroup(hosts)` is confusing: the constructor takes the name of a tool group as argument, not a host. Even though it is not called at all, it should still make sense.